packages/react-server/src/ReactFlightServer.js

```diff
@@ -1911,7 +1921,13 @@ function renderModelDestructive(
 
     const iteratorFn = getIteratorFn(value);
     if (iteratorFn) {
-      return renderFragment(request, task, Array.from((value: any)));
+      // TODO: Should we serialize the return value as well like we do for AsyncIterables?
```

Comment:
This is kind of annoying because it would require more code. We could potentially just not support it in the AsyncIterable case too.